treeherder/webapp/api/groups.py

```diff
@@ -44,64 +42,71 @@ def list(self, request):
         if (enddate - startdate).days > 1:
             enddate = startdate + datetime.timedelta(days=1)
 
-        q = (
-            Job.objects.filter(
-                push__time__gte=str(startdate.date()), push__time__lte=str(enddate.date())
+        self.queryset = (
+            JobLog.objects.filter(
+                job__push__time__gte=str(startdate.date()), job__push__time__lte=str(enddate.date())
             )
-            .filter(repository_id__in=(1, 77))
-            .values(
-                'job_log__groups__name',
-                'job_type__name',
-                'job_log__group_result__status',
-                'failure_classification_id',
+            .values('job_id')
+            .filter(
+                job__repository_id__in=(1, 77),
+                job__job_type__name__startswith='test-',
+                group_result__status__in=(1, 2),
             )
-            .annotate(job_count=Count('id'))
-            .order_by('job_log__groups__name')
+            .exclude(
+                groups__name='',
+            )
+            .annotate(
+                job_count=Count('job_id'),
+                result=Case(
+                    When(group_result__status=1, then=Value("passed")),
+                    When(group_result__status=2, then=Value("testfailed")),
```

Comment:
what happens when we get a different value instead of 1 || 2 ?

Reply:
That is a good point. Django will automatically add a `ELSE NULL` in the CASE clause, but no null value should be returned as it only keep entries with `group_result__status__in=(1, 2),` (in the `filter` just above).
I'll add a little comment for this.